File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.js
function solution(new_id) {
    reg1 = /[^0-9a-z-_.]/g
    reg2 = /\.{2,}/g
    reg3 = /^\.+|\.$/g
    
    new_id=new_id.toLowerCase().replace(reg1,"").replace(reg2, ".").replace(reg3, "")
    
    if (new_id === "") new_id = "a"
    if (new_id.length >= 16) new_id = new_id.slice(0,15).replace(/\.$/, "")
    if (new_id.length < 3) new_id += new_id[new_id.length-1].repeat(3-new_id.length)
    
    return new_id;
}

File: 프로그래머스/1/72410. 신규 아이디 추천/신규 아이디 추천.py
import re

def solution(new_id):
    new_id = re.sub('[^0-9a-z\-._]', '', new_id.lower())
    new_id = re.sub('\.{2,}', '.', new_id)
    new_id = re.sub('^[.]|[.]$', '', new_id)
    
    if new_id == "": new_id = "a"
    if len(new_id) >= 16:
        new_id = re.sub('[.]$', '', new_id[:15])
    if len(new_id) < 3: new_id += new_id[-1] * (3-len(new_id))
    
    return new_id